Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -758,19 +807,58 @@ void doEsirkepovDepositionShapeN (const GetParticlePosition<PIdx>& GetPosition,
 #if !defined(WARPX_DIM_1D_Z)
             double sx_new[depos_order + 3] = {0.};
             double sx_old[depos_order + 3] = {0.};
-            const int i_new = compute_shape_factor(sx_new+1, x_new);
+            const int i_new = compute_shape_factor(sx_new+1, x_new );
             const int i_old = compute_shifted_shape_factor(sx_old, x_old, i_new);
+            // If particle is close to the embedded boundary, recompute deposition with order 1 shape
+            if constexpr (reduce_shape_control == has_reduced_shape) {
+                if (reduce_shape_new) {
+                    for (int i=0; i<depos_order+3; i++) sx_new[i] = 0.; // Erase previous deposition
```

Comment:
I took a look on godbolt and the compiler seems to unroll this constant-size loop, so no need to write it more obscurely.